Source/NSPathCell.m

```diff
@@ -0,0 +1,309 @@
+/* Implementation of class NSPathCell
+   Copyright (C) 2020 Free Software Foundation, Inc.
+   
+   By: Gregory John Casamento
+   Date: Wed Apr 22 18:19:07 EDT 2020
+
+   This file is part of the GNUstep Library.
+   
+   This library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+   
+   This library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+   
+   You should have received a copy of the GNU Lesser General Public
+   License along with this library; if not, write to the Free
+   Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
+   Boston, MA 02110 USA.
+*/
+
+#import "AppKit/NSPathCell.h"
+#import "AppKit/NSWorkspace.h"
+#import "AppKit/NSImage.h"
+#import "AppKit/NSPathComponentCell.h"
+
+@interface NSPathCell (Private)
++ (NSArray *) _generateCellsForURL: (NSURL *)url;
+@end
+
+@interface NSPathComponentCell (Private)
+- (void) _setLastComponent: (BOOL)f;
+@end
+
+@implementation NSPathCell
+
+- (void)mouseEntered:(NSEvent *)event 
+           withFrame:(NSRect)frame 
+              inView:(NSView *)view
+{
+}
+
+- (void)mouseExited:(NSEvent *)event 
+          withFrame:(NSRect)frame 
+             inView:(NSView *)view
+{
+}
+
+- (void) setAllowedTypes: (NSArray *)types
+{
+  ASSIGNCOPY(_allowedTypes, types);
+}
+
+- (NSArray *) allowedTypes
+{
+  return _allowedTypes;
+}
+
+- (NSPathStyle) pathStyle
+{
+  return _pathStyle;
+}
+
+- (void) setPathStyle: (NSPathStyle)pathStyle
+{
+  _pathStyle = pathStyle;
+}
+
+- (void) setControlSize: (NSControlSize)size
+{
+  _controlSize = size;
+}
+
+- (void) setObjectValue: (id)obj
+{
+  ASSIGN(_objectValue, obj);
```

Comment:
We already know that this is the url, so we should call setURL: here. Doing so will resolve the XIB decoding issue.

Reply:
As I said, I cannot seem to get NSContents to unarchive.